packages/wallet/src/Wallets/BaseWallet.ts

```diff
@@ -113,6 +121,7 @@ export interface BaseWalletProps {
   readonly name: string;
   readonly polling?: PollingConfig;
   readonly retryBackoffConfig?: RetryBackoffConfig;
+  readonly purpose: KeyPurpose;
```

Comment:
`BaseWallet` shouldn't know about purpose, because it's not a bip32 wallet, it can also be a script wallet. `purpose` should probably only be passed to key management utils like the KeyAgent.

We might need to add optional `stakeKeyPurpose`/`paymentKeyPurpose` to [GroupedAddress](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/key-management/src/types.ts#L72).

@AngelCastilloB I think we should consider refactoring GroupedAddress to better model the addresses, maybe group some bip32 properties. WDYT?
```suggestion
```

Reply:
Removed `purpose` from BaseWallet on 3487275